amd_openvx_extensions/amd_rpp/source/tensor/MelFilterBank.cpp

```diff
@@ -0,0 +1,252 @@
+/*
+Copyright (c) 2019 - 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "internal_publishKernels.h"
+
+struct MelFilterBankLocalData {
+    vxRppHandle *handle;
+    Rpp32u deviceType;
+    RppPtr_t pSrc;
+    RppPtr_t pDst;
+    Rpp32f freqHigh;
+    Rpp32f freqLow;
+    RpptMelScaleFormula melFormula;
+    Rpp32s nfilter;
+    bool normalize;
+    Rpp32f sampleRate;
+    RpptDescPtr pSrcDesc;
+    RpptDescPtr pDstDesc;
+    RpptImagePatch *pSrcDims;
+    size_t inputTensorDims[RPP_MAX_TENSOR_DIMS];
+    size_t outputTensorDims[RPP_MAX_TENSOR_DIMS];
+};
+
+void copy_src_dims_and_update_dst_roi(MelFilterBankLocalData *data, RpptROI *src_roi, RpptROI *dst_roi) {
+    for (unsigned i = 0; i < data->inputTensorDims[0]; i++) {
+       dst_roi[i].xywhROI.xy.x = src_roi[i].xywhROI.xy.x;
+       dst_roi[i].xywhROI.xy.y = data->nfilter;
+       data->pSrcDims[i].width = src_roi[i].xywhROI.xy.x;
+       data->pSrcDims[i].height = src_roi[i].xywhROI.xy.y;
+    }
+}
+
+static vx_status VX_CALLBACK refreshMelFilterBank(vx_node node, const vx_reference *parameters, vx_uint32 num, MelFilterBankLocalData *data) {
+    vx_status status = VX_SUCCESS;
+    void *roi_tensor_ptr_src, *roi_tensor_ptr_dst;
+    if (data->deviceType == AGO_TARGET_AFFINITY_GPU) {
+#if ENABLE_OPENCL
+        return VX_ERROR_NOT_IMPLEMENTED;
+#elif ENABLE_HIP
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_BUFFER_HIP, &data->pSrc, sizeof(data->pSrc)));
```

Comment:
Please return VX_ERROR_NOT_IMPLEMENTED;

Reply:
I think it is not required to add this here, because we are just querying the params for input, output and ROI
irrespective of backend this should work. In processGraph we already added this VX_ERROR_NOT_IMPLEMENTED error in process